lib/bing_translator.rb

```diff
@@ -16,18 +16,21 @@ class BingTranslator
   NAMESPACE_URI = 'http://api.microsofttranslator.com/V2'
   ACCESS_TOKEN_URI = 'https://datamarket.accesscontrol.windows.net/v2/OAuth2-13'
   DATASETS_URI = "https://api.datamarket.azure.com/Services/My/Datasets?$format=json"
+  COGNITIVE_ACCESS_TOKEN_URI = 'https://api.cognitive.microsoft.com/sts/v1.0/issueToken'
 
   class Exception < StandardError; end
   class AuthenticationException < StandardError; end
 
-  def initialize(client_id, client_secret, skip_ssl_verify = false, account_key = nil)
+  def initialize(client_id: nil, client_secret: nil, skip_ssl_verify: false, account_key: nil, subscription_key: nil)
```

Comment:
This seems like an unnecessary breaking change. Named parameters seem cleaner, but I'm not sure it's worth forcing downstream users to change their code.

If we do choose to do this, though, I think we'd want to make `client_id` and `client_secret` *required* named arguments:
```ruby
def initialize(client_id:, client_secret:, skip_ssl_verify: false, account_key: nil, subscription_key: nil)
```

Also, the `README.md` examples need to be updated to match changes here.